lms/templates/courseware/dates.html

```diff
@@ -42,27 +62,20 @@ <h2 class="hd hd-2 date-title">
                                             % if active:
                                                 <div class="pill due">${_('Due Today')}</div>
                                             % endif
-                                            % if block_is_verified and not learner_is_verified:
+                                            % if not learner_has_access:
                                                 <div class="pill verified"><span class="fa fa-lock verified-icon" aria-hidden="true"></span>${_('Verified Only')}</div>
                                             % endif
                                         % endif
                                 </div>
-                                <div class="timeline-title ">
-                                    % if block.title_html and is_assignment:
+                                <div class="timeline-title ${access_class}">
+                                    % if block.title_html and is_assignment and learner_has_access:
```

Comment:
If the learner doesn't have access, they shouldn't see the text title either, right? Also, why is `is_assignment` checked here? I think I'm a bit confused about what this condition is for.

Reply:
The choice here is whether to link to the unit or not. So I assume the assignment check there is that we only want to link to assignments, not other kinds of blocks? (This conditional predates me though.)

As for showing the title if they don't have access -- I think the idea is that we want to show them what they could be getting if they upgraded.